### PR TITLE
Add running tests to developer contribution section #322

### LINKED ISSUE #322: Add running tests to developer contribution section
Although we now have a fully update source build and basic functional test guide in contribute.rst via now merged:
#321 
We do not have any mention or guide on running our build-in unit tests.

It is proposed that we add a testing section to our "Build VM initial setup"  so that we have a new 3. entry:

1. configure
2. build
3. test

And move the existing text concerning manual testing into this proposed 3rd section. And add to it info on our existing tests and how to run them:
```
cd /opt/rockstor-dev/
./bin/test -v 2

Ran 216 tests in 17.094s

FAILED (errors=1)
```
(See https://github.com/rockstor/rockstor-core/issues/2327 for our current single test error)

### PR DESCRIPTION
**Pull request by phillxnet**
Add a testing section to our "Build VM initial setup" so that we have a new 3. entry thus:

1. configure
2. build
3. test

And move the existing text concerning manual testing into this proposed 3rd section. Add info on our existing tests and how to run them with an eye to our issues for outstanding failures.

Fixes #322
@FroggyFlox Ready for review.

### Checklist
- [x] With the proposed changes no Sphinx errors or warnings are generated.
- [x] I have added my name to the AUTHORS file, if required (descending alphabetical order).

### CODE REVIEW

File: contribute/contribute.rst [hunk 1]
```diff
@@ -320,6 +327,23 @@ Be sure to check that the resulting build behaves as expected over:
 In (1.) above a db wipe is initiated helping to test the self-start code capability.
 See the initrock script and it's systemd trigger service: rockstor-init.
 
+We also have **automated tests** in place that cover our API's and core critical path functionality.
+It is expected that any changes to critical path code e.g. fs management / updates / Web-UI / services;
```
Comment (FroggyFlox) [inline contribute/contribute.rst:331]: Would a comma be preferred here?
```suggestion
It is expected that any changes to critical path code e.g. fs management / updates / Web-UI / services,
```

File: contribute/contribute.rst [hunk 2]
```diff
@@ -320,6 +327,23 @@ Be sure to check that the resulting build behaves as expected over:
...
+include counterpart contributions to prove the expected function, if required.
+This is an oft neglected element in software development;
+but we are attempting to better our own standing in this regard.
+
+The following will run all tests post the above detailed source install:
```
Comment (FroggyFlox) [inline contribute/contribute.rst:336]: Style suggestion:
```suggestion
The following will run all tests following the source installation detailed above:
```

**Review by FroggyFlox — APPROVED** [review]
Thanks a lot, @phillxnet !

I only have a couple minor style edits for you to include if you'd like (optional).

File: contribute/contribute.rst (hunk 1 above)
Comment (phillxnet) [reply contribute/contribute.rst:331]: I can go with that; but I do like my semi-colons.

File: contribute/contribute.rst (hunk 2 above)
Comment (phillxnet) [reply contribute/contribute.rst:336]: OK; we can pull off 2 x 'following' in the same sentence :). And reads better overall anyway. Thanks.

**Comment by phillxnet** [comment]
@FroggyFlox Thanks for the review. Much appreciated.
PR product now in production.

**Pull request merged**